### ISSUE TITLE
`<MockBackend as Backend>::Error` should be a real error type

### ISSUE THREAD

**Issue by a1phyr** [tesselode/kira#48]
Currently, `<MockBackend as Backend>::Error` is `()` so it cannot be converted to `Box<dyn std::error::Error>`.

Using `std::convert::Infallible` or a standalone type that implement `std::error::Error` would be more convenient.

**Comment by tesselode** [comment]
Are you making a library based on Kira?

**Comment by a1phyr** [comment]
Yes I was making [`asset_manager-kira`](https://lib.rs/crates/assets_manager-kira), an integration with [`assets_manager`](https://lib.rs/crates/assets_manager)
